infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -113,20 +114,23 @@ boolean checkBootstrap(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r)
     // Helper Methods
     @ServerHandler(type=CorfuMsgType.LAYOUT_REQUEST)
     public synchronized void handleMessageLayoutRequest(CorfuPayloadMsg<Long> msg, ChannelHandlerContext ctx, IServerRouter r) {
-        if (!checkBootstrap(msg, ctx, r)) { return; }
-        long epoch = msg.getPayload();
-        if (epoch <= serverContext.getServerEpoch()) {
-            r.sendResponse(ctx, msg, new LayoutMsg(getCurrentLayout(), CorfuMsgType.LAYOUT_RESPONSE));
-            return;
-        } else {
-            // else the client is somehow ahead of the server.
-            //TODO figure out a strategy to deal with this situation
-            // Very odd ... if we don't send any response here, we hang the OTP mailbox thread.
-            long serverEpoch = serverContext.getServerEpoch();
-            r.sendResponse(ctx, msg, new CorfuPayloadMsg<>(CorfuMsgType.WRONG_EPOCH, serverEpoch));
-            log.warn("Message Epoch {} ahead of Server epoch {}", epoch, serverContext.getServerConfig());
+        Timer.Context context = CorfuServer.timerLayoutReq.time();
```

Comment:
In the afternoon's discussion, will unindent & use function wrapper.  Not discussed, by my preference is to add additional interrupted execution counters if desired later. A try/finally will permit the existing metric to update as an exception is thrown past/by these points.